wgpu-hal/src/vulkan/device.rs

```diff
@@ -699,6 +697,27 @@ impl super::Device {
     pub fn raw_device(&self) -> &ash::Device {
         &self.shared.raw
     }
+
+    pub fn enabled_device_extensions(&self) -> &[&'static CStr] {
+        &self.shared.enabled_extensions
+    }
+
+    pub fn raw_physical_device(&self) -> ash::vk::PhysicalDevice {
+        self.shared.physical_device
+    }
+
+    pub fn raw_instance(&self) -> &ash::Instance {
+        &self.shared.instance.raw
+    }
+
+    /// Returns the driver api version of the instance.
+    pub fn driver_api_version(&self) -> u32 {
+        self.shared.instance.driver_api_version
+    }
+
+    pub fn enabled_instance_extensions(&self) -> &[&'static CStr] {
```

Comment:
This feels weird, considering you can just call this function on the instance.

Reply:
Yes that could be done. However the wgpu apis implicitly hide the instance when the device is created.

One option could be to make the underlying InstanceShared accessible so we don't need to duplicate so many functions for instance fields.